### PR TITLE
Adding keyboard shortcuts for column resizing in slickgrid

### PR DESCRIPTION
**Pull request by aasimkhan30**
Since the headers are not focusable, we are trying to provide alternatives to perform header based actions using the keyboard. 

azuredatastudio_c1dP16rtOm

Issue:
#20779

### CODE REVIEW

File: src/sql/base/browser/ui/table/table.ts [hunk 1]
```diff
@@ -132,6 +132,24 @@ export class Table<T extends Slick.SlickData> extends Widget implements IDisposa
 
 		this._grid.onKeyDown.subscribe((e, args: Slick.OnKeyDownEventArgs<T>) => {
 			const evt = (e as JQuery.TriggeredEvent).originalEvent as KeyboardEvent;
+			if (evt.altKey && [37, 39].includes(evt.keyCode)) {
```
Comment (Charles-Gagnon) [inline src/sql/base/browser/ui/table/table.ts:135]: Use keyCodes from src\vs\base\common\keyCodes.ts

File: src/sql/base/browser/ui/table/table.ts [hunk 2]
```diff
@@ -132,6 +132,24 @@ export class Table<T extends Slick.SlickData> extends Widget implements IDisposa
...
+
+				const activeCell = this._grid.getActiveCell();
+				if (activeCell) {
+					const columns = this._grid.getColumns();
+					if (evt.keyCode === 37) {
+						if (columns[activeCell.cell].width > 10) {
+							columns[activeCell.cell].width -= 10;
```
Comment (Charles-Gagnon) [inline src/sql/base/browser/ui/table/table.ts:142]: Why 10? Are you trying to simulate dragging the columns to change the width? (comments for intended functionality would be nice...)

Is that a common pattern you've seen elsewhere?

**Review by Charles-Gagnon — COMMENTED** [review]
How is this addressing the linked issue? As far as I can tell this isn't changing anything about being able to navigate to the headers.

File: src/sql/base/browser/ui/table/table.ts (hunk 1 above)
Comment (alanrenmsft) [inline src/sql/base/browser/ui/table/table.ts:135]: as always, please add comment explaining the purpose and the future improvements.

File: src/sql/base/browser/ui/table/table.ts (hunk 2 above)
Comment (Charles-Gagnon) [reply src/sql/base/browser/ui/table/table.ts:144]: Sure, you should leave a comment about that - magic numbers are bad

Was this feature something you came up with yourself or is there a standard you're following here?

Comment (aasimkhan30) [reply src/sql/base/browser/ui/table/table.ts:151]: I have come up with this feature. I am not following any existing standard on it.

**Comment by alanrenmsft** [comment]
ALT left/right? I believe it is used already.

I thought about it again after our chat, maybe we should copy the excel's behavior and prompt an input box for user to type in the width. the input box should have the current width populated by default. with this approach, you will need another parameter to pass in an input provider, similar to what you did for accessibility service.

**Comment by aasimkhan30** [comment]
@alanrenmsft , I do not think we are using alt left/right anywhere else. Moreover, this keyboard shortcut will just be limited to the grid so we are not hampering any other shortcuts for now.

**Comment by alanrenmsft** [comment]
I would agree if it is another keybinding, but ALT left/right is to navigation, you can open multiple documents and try it, or you can try it in vscode.

**Comment by aasimkhan30** [comment]
@alanrenmsft , can you please suggest an alternate keybinding?

**Comment by alanrenmsft** [comment]
maybe "shift+alt+s"? it is not used.

**Comment by alanrenmsft** [comment]
1. update pr description
2. a gif demonstrating the feature

File: src/sql/base/browser/ui/table/table.ts [hunk 3]
```diff
@@ -135,8 +144,50 @@ export class Table<T extends Slick.SlickData> extends Widget implements IDisposa
...
+			 */
+			const stdEvt = new StandardKeyboardEvent(evt);
+
+			if (stdEvt.altKey && stdEvt.shiftKey && stdEvt.keyCode === KeyCode.KeyS) {
+
+				const activeCell = this._grid.getActiveCell();
+				if (activeCell) {
+					const columns = this._grid.getColumns();
+					/**
+					 * increasing/decreasing the column width by 10px as 1px will be too slow to make
```
Comment (alanrenmsft) [inline src/sql/base/browser/ui/table/table.ts:164]: comments are no longer relevant.

File: src/sql/base/browser/ui/table/table.ts [hunk 4]
```diff
@@ -135,8 +144,50 @@ export class Table<T extends Slick.SlickData> extends Widget implements IDisposa
...
+			 * Adding a keyboard shortcut Alt+left/right arrow to change column width. This is a
+			 * workaround to provide an alternative to column header mouse drag based resizing.
+			 * TODO: Make the shortcut configurable or provide a way to focus on column headers
+			 * so that users can focus a header and then use the arrow keys to resize columns.
+			 */
+			const stdEvt = new StandardKeyboardEvent(evt);
+
+			if (stdEvt.altKey && stdEvt.shiftKey && stdEvt.keyCode === KeyCode.KeyS) {
+
+				const activeCell = this._grid.getActiveCell();
```
Comment (alanrenmsft) [inline src/sql/base/browser/ui/table/table.ts:160]: create a method to handle the change width logic

File: src/sql/base/browser/ui/table/table.ts [hunk 5]
```diff
@@ -135,8 +144,50 @@ export class Table<T extends Slick.SlickData> extends Widget implements IDisposa
...
+
+			/**
+			 * Adding a keyboard shortcut Alt+left/right arrow to change column width. This is a
+			 * workaround to provide an alternative to column header mouse drag based resizing.
+			 * TODO: Make the shortcut configurable or provide a way to focus on column headers
+			 * so that users can focus a header and then use the arrow keys to resize columns.
+			 */
+			const stdEvt = new StandardKeyboardEvent(evt);
+
+			if (stdEvt.altKey && stdEvt.shiftKey && stdEvt.keyCode === KeyCode.KeyS) {
```
Comment (alanrenmsft) [inline src/sql/base/browser/ui/table/table.ts:158]: I think there is a resizable option for each column, make sure you respect that option.

**Review by alanrenmsft — APPROVED** [review]

**Pull request merged**